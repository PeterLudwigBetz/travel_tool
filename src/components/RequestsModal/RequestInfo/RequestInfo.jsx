
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TravelLink from '../_RequestTravel';
import generateDynamicDate from '../../../helper/generateDynamicDate';

class RequestInfo extends Component {
  render() {
    const { requestData } = this.props;
    const { createdAt, departureDate, arrivalDate } = requestData;
    return (
      <div className="modal__modal-date">
        <TravelLink
          divClass="modal__travel-date" innerClass="modal__travel-dates"
          dynamicText="Date submitted" nextClass="modal__date-text" dynamicDate={generateDynamicDate(requestData, createdAt)} />
        <TravelLink
          divClass="modal__travel-date" innerClass="modal__travel-dates"
          dynamicText="Target depature date" nextClass="modal__date-text" dynamicDate={generateDynamicDate(requestData, departureDate)} />

        <TravelLink
          divClass="modal__travel-date" innerClass="modal__travel-dates"
          dynamicText="Target return date" nextClass="modal__date-text" dynamicDate={generateDynamicDate(requestData, arrivalDate)} />
      </div>
    );
  }
}

RequestInfo.propTypes = {
  requestData: PropTypes.object,
};

RequestInfo.defaultProps = {
  requestData: {},
};

export default RequestInfo;
