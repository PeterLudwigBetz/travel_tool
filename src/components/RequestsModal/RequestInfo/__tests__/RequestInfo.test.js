import React from 'react';
import { shallow } from 'enzyme';
import RequestInfo from '../RequestInfo';
import requestData from '../../__mocks__/requestData';

const props = {
  requestData,
};

describe('UserComments component', () => {
  const wrapper = shallow(<RequestInfo {...props} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the UserComments component as expected', () => {
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('TravelLink').length).toBe(3);
    expect(wrapper.length).toBe(1);
  });
});
