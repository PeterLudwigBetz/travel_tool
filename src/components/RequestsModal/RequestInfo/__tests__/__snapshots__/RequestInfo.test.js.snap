// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComments component should match snapshot 1`] = `
<div
  className="modal__modal-date"
>
  <TravelLink
    divClass="modal__travel-date"
    dynamicDate="04 Sep 2018"
    dynamicText="Date submitted"
    innerClass="modal__travel-dates"
    nextClass="modal__date-text"
  />
  <TravelLink
    divClass="modal__travel-date"
    dynamicDate="12 Sep 2018"
    dynamicText="Target depature date"
    innerClass="modal__travel-dates"
    nextClass="modal__date-text"
  />
  <TravelLink
    divClass="modal__travel-date"
    dynamicDate="12 Nov 2018"
    dynamicText="Target return date"
    innerClass="modal__travel-dates"
    nextClass="modal__date-text"
  />
</div>
`;
